crates/miden-lib/src/account/interface/test.rs

```diff
@@ -207,6 +207,30 @@ fn test_custom_account_default_note() {
     );
 }
 
+/// Checks the function `create_swap_note` should fail if the requested asset is the same as the
+/// offered asset.
+#[test]
+fn test_required_asset_same_as_offered() {
+    let offered_asset = NonFungibleAsset::mock(&[1, 2, 3, 4]);
+    let requested_asset = NonFungibleAsset::mock(&[1, 2, 3, 4]);
+
+    let result = create_swap_note(
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_IMMUTABLE_CODE.try_into().unwrap(),
+        offered_asset,
+        requested_asset,
+        NoteType::Public,
+        ZERO,
+        NoteType::Public,
+        ZERO,
+        &mut RpoRandomCoin::new(Word::from([1, 2, 3, 4u32])),
+    );
+
+    assert!(matches!(
+        result,
+        Err(NoteError::Other { error_msg, .. }) if error_msg == "requested asset same as offered asset".into()
+    ));
```

Comment:
For error matching we usually use `assert_matches!` macro, I think it should be ok to use here as well (rather than on the string)